huaweicloud/services/rms/resource_huaweicloud_rms_organizational_policy_assignment.go

```diff
@@ -0,0 +1,433 @@
+	} else {
+		res["custom_policy_assignment_metadata"] = metadata
+	}
+
+	return res, nil
+}
+
+func resourceOrganizationalPolicyAssignmentCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createOrganizationalPolicyAssignment: Create a RMS organizational policy assignment.
+	var (
+		createOrgPolicyAssignmentHttpUrl = "v1/resource-manager/organizations/{organization_id}/policy-assignments"
+		createOrgPolicyAssignmentProduct = "rms"
+	)
+	createOrgPolicyAssignmentClient, err := cfg.NewServiceClient(createOrgPolicyAssignmentProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RMS client: %s", err)
+	}
+
+	createOrgPolicyAssignmentPath := createOrgPolicyAssignmentClient.Endpoint + createOrgPolicyAssignmentHttpUrl
+	createOrgPolicyAssignmentPath = strings.ReplaceAll(createOrgPolicyAssignmentPath, "{organization_id}", d.Get("organization_id").(string))
+
+	createOrgPolicyAssignmentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	createOpts, err := buildOrganizationalPolicyAssignmentCreateOpts(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createOrgPolicyAssignmentOpt.JSONBody = utils.RemoveNil(createOpts)
+	createOrgPolicyAssignmentResp, err := createOrgPolicyAssignmentClient.Request("PUT",
+		createOrgPolicyAssignmentPath, &createOrgPolicyAssignmentOpt)
+	if err != nil {
+		return diag.Errorf("error creating RMS organizational policy assignment: %s", err)
+	}
+
+	createOrgPolicyAssignmentRespBody, err := utils.FlattenResponse(createOrgPolicyAssignmentResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if d.IsNewResource() {
+		id, err := jmespath.Search("organization_policy_assignment_id", createOrgPolicyAssignmentRespBody)
+		if err != nil {
+			return diag.Errorf("error creating RMS assignment package: ID is not found in API response")
+		}
+		d.SetId(id.(string))
+	}
+
+	return resourceOrganizationalPolicyAssignmentRead(ctx, d, meta)
+}
+
+func flattenOrganizationalPolicyFilter(filter interface{}) []map[string]interface{} {
+	if filter == nil {
+		return nil
+	}
+
+	filterMap := filter.(map[string]interface{})
+	return []map[string]interface{}{
+		{
+			"region":            filterMap["region"],
+			"resource_provider": filterMap["resource_provider"],
+			"resource_type":     filterMap["resource_type"],
+			"resource_id":       filterMap["resource_id"],
+			"tag_key":           filterMap["tag_key"],
+			"tag_value":         filterMap["tag_value"],
+		},
+	}
+}
+
+func flattenOrganizationalPolicyParameters(parameters interface{}) (map[string]interface{},
+	error) {
+	if parameters == nil {
+		return nil, nil
+	}
+
+	result := make(map[string]interface{})
+	for k, v := range parameters.(map[string]interface{}) {
+		val := v.(map[string]interface{})
+		jsonBytes, err := json.Marshal(val["value"])
+		if err != nil {
+			return nil, fmt.Errorf("generate json string failed: %s", err)
```

Comment:
it's better to add the key in error message
```suggestion
			return nil, fmt.Errorf("generate json string of %s failed: %s", key, err)
```

Reply:
fixed